### PR TITLE
feat(RDS): add rds pg account resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add rds pg account resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add rds pg account resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds/' TESTARGS='-run TestAccPgAccount_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccPgAccount_basic -timeout 360m -parallel 4 
=== RUN   TestAccPgAccount_basic 
=== PAUSE TestAccPgAccount_basic
=== CONT  TestAccPgAccount_basic
--- PASS: TestAccPgAccount_basic (632.25s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       632.294s
```

### CODE REVIEW

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	createPgAccountPath = strings.ReplaceAll(createPgAccountPath, "{project_id}", createPgAccountClient.ProjectID)
+	createPgAccountPath = strings.ReplaceAll(createPgAccountPath, "{instance_id}", instanceId)
+
+	createPgAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	requestBody := buildCreatePgAccountBodyParams(d)
+	log.Printf("[DEBUG] Create RDS PostgreSQL account options: %#v", createPgAccountOpt)
+	requestBody["password"] = utils.ValueIngoreEmpty(d.Get("password"))
```
Comment (hmbig2) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go]:
```suggestion
```
set password in func: buildCreatePgAccountBodyParams

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	return getPgAccountRespBody, nil
+}
+
+func TestAccPgAccount_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	updateName := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_rds_pg_account.test"
+	pwd := fmt.Sprintf("%s%s%d", acctest.RandString(5), acctest.RandStringFromCharSet(2, "!#%^*"),
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]:
```suggestion
	pwd := utils.randompassword()
```
now this function has met the rule

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+%s
+
+resource "huaweicloud_rds_pg_account" "test" {
+  instance_id = huaweicloud_rds_instance.test.id
+  name        = "%s"
+  password    = "Test@123456789"
+  description = "test_description_update"
+}
+`, testAccRdsInstance_basic(name, dbPwd), name)
+}
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]:
```suggestion
func testPgAccount_basic(name, dbPwd,accountPwd, description string) string {
	return fmt.Sprintf(`
%s

resource "huaweicloud_rds_pg_account" "test" {
  instance_id = huaweicloud_rds_instance.test.id
  name        = "%s"
  password    = "Test@12345678"
  description = "test_description"
}
`, testAccRdsInstance_basic(name, dbPwd), name,accountPwd, description)
}
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go]: no, the password should not be print in the log

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]: it has been changed to a fixed value

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]: done

Comment (houpeng80) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]: make testacc TEST=./huaweicloud/services/acceptance/rds/ TESTARGS='-run TestAccPgAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccPgAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccPgAccount_basic
=== PAUSE TestAccPgAccount_basic
=== CONT  TestAccPgAccount_basic
--- PASS: TestAccPgAccount_basic (624.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       624.415s

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	)
+	getPgAccountClient, err := cfg.NewServiceClient(getPgAccountProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating RDS client: %s", err)
+	}
+
+	// Split instance_id and user from resource id
+	parts := strings.Split(state.Primary.ID, "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid id format, must be <instance_id>/<name>")
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go:34]:
```suggestion
		return nil, fmt.Errorf("invalid ID format, must be <instance_id>/<name>")
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+		return nil, fmt.Errorf("error retrieving RDS PostgreSQL account: %s", err)
+	}
+
+	account := utils.PathSearch(fmt.Sprintf("users[?name=='%s']|[0]", accountName), getPgAccountRespBody, nil)
+
+	if account != nil {
+		return account, nil
+	}
+
+	return nil, fmt.Errorf("error get RDS PostgreSQL account by instanceID %s and account %s", instanceId, accountName)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go:69]:
```suggestion
	return nil, fmt.Errorf("error getting RDS PostgreSQL account by instanceID %s and account %s", instanceId, accountName)
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getPgAccountResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getPgAccount: query RDS PostgreSQL account
+	var (
+		getPgAccountHttpUrl = "v3/{project_id}/instances/{instance_id}/db_user/detail?page=1&limit=100"
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go:23]: I think the query parameters is useless here

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "instance_id",
+						"huaweicloud_rds_instance.test", "id"),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "description", "test_description"),
+				),
+			},
+			{
+				Config: testPgAccount_basic(updateName, "Test@123456789", "test_description_update"),
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]: Field `name` is `ForceNew`, do not check editing a `ForceNew` field.

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+	createPgAccountPath = strings.ReplaceAll(createPgAccountPath, "{project_id}", createPgAccountClient.ProjectID)
+	createPgAccountPath = strings.ReplaceAll(createPgAccountPath, "{instance_id}", instanceId)
+
+	createPgAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	requestBody := buildCreatePgAccountBodyParams(d)
+	log.Printf("[DEBUG] Create RDS PostgreSQL account options: %#v", requestBody)
+	requestBody["password"] = d.Get("password")
```
Comment (deer-hang) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go:96]: This line can remove to function `buildCreatePgAccountBodyParams(d)`

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+	updatePgAccountPath = strings.ReplaceAll(updatePgAccountPath, "{project_id}", client.ProjectID)
+	updatePgAccountPath = strings.ReplaceAll(updatePgAccountPath, "{instance_id}", d.Get("instance_id").(string))
+
+	updatePgAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	updatePgAccountOpt.JSONBody = utils.RemoveNil(buildUpdatePgAccountPasswordBodyParams(d))
+
+	instanceId := d.Get("instance_id").(string)
```
Comment (deer-hang) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go]: Repeat field definition. I think this line can move to 219.

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go (hunk 9 above)
Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go]: done

**Comment by houpeng80** [comment]
make testacc TEST=./huaweicloud/services/acceptance/rds/ TESTARGS='-run TestAccPgAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccPgAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccPgAccount_basic
=== PAUSE TestAccPgAccount_basic
=== CONT  TestAccPgAccount_basic
--- PASS: TestAccPgAccount_basic (801.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       801.651s

**Pull request merged**